src/loader/lwlocks.c

```diff
@@ -0,0 +1,54 @@
+/*
+ * This file and its contents are licensed under the Apache License 2.0.
+ * Please see the included NOTICE for copyright information and
+ * LICENSE-APACHE for a copy of the license.
+ */
+
+#include <postgres.h>
+#include <fmgr.h>
+#include <miscadmin.h>
+#include <storage/lwlock.h>
+#include <storage/shmem.h>
+
+#include "loader/lwlocks.h"
+
+#define TS_LWLOCK_SHM_NAME "ts_lwlock_shm_name"
+#define CHUNK_APPEND_LWLOCK_TRANCHE_NAME "ts_chunk_append_lwlock_tranche"
+
+/*
+ * since shared memory can only be setup in a library loaded as
+ * shared_preload_libraries we have to setup this struct here
+ */
+typedef struct TSLWLocks
+{
+	LWLock *chunk_append;
+} TSLWLocks;
+
+static TSLWLocks *ts_lwlocks = NULL;
+
+void
+ts_chunk_append_shmem_startup()
+{
+	bool found;
+	LWLock **lock_pointer;
+
+	LWLockAcquire(AddinShmemInitLock, LW_EXCLUSIVE);
+	ts_lwlocks = ShmemInitStruct(TS_LWLOCK_SHM_NAME, sizeof(TSLWLocks), &found);
+	if (!found)
+	{
+		memset(ts_lwlocks, 0, sizeof(TSLWLocks));
+		ts_lwlocks->chunk_append = &(GetNamedLWLockTranche(CHUNK_APPEND_LWLOCK_TRANCHE_NAME))->lock;
+	}
+	LWLockRelease(AddinShmemInitLock);
+
+	/* assign to rendezvous variable so it can be found by other libraries */
```

Comment:
would be nice to add explanation for why this variable is to specific locks and not to struct